src/scheduler/Scheduler.java

```diff
@@ -69,12 +68,15 @@ private void executeProcess(PCB processPCB, boolean roundRobin) {
                 if (processIO) {
                     process.setState(State.BLOCKED);
                     processPCB.setWaitTo2();
+                    ProcessList.decreaseBlockedListWait(roundRobin);
                     ProcessList.addBlockedProcess(processPCB);
                 } else {
                     if (roundRobin) {
                         process.setState(State.READY);
                         ProcessList.addReadyProcessInLastPosition(processPCB);
+                        ProcessList.decreaseBlockedListWait(roundRobin);
                     } else {
+                        ProcessList.decreaseBlockedListWait(roundRobin);
```

Comment:
By your logic, if a process ends, blocked list will not decrease wait of blocked process. This should be moved to last line before end of `do...while()`.